backend/experiment/utils.py

```diff
@@ -38,23 +46,58 @@ def slugify(text):
     return text.lower()
 
 
-def non_breaking_spaces(s):
-    # Convert regular spaces to non breaking spacing
-    # on the given string
+def non_breaking_spaces(s) -> str:
```

Comment:
perhaps we can give this argument a more descriptive name, e.g., `input_string`.